### PR TITLE
Organization management

### PR DESCRIPTION
**Pull request by DashaBoichuk**
Implemented Services, Dto, Mappers

### CODE REVIEW

File: src/main/java/com/ita/if103java/ims/service/impl/MailServiceImpl.java [hunk 1]
```diff
@@ -40,4 +40,19 @@ public void sendMessage(UserDto userDto, String message, String subject) {
...
+        try {
+            mimeMessage.addRecipient(Message.RecipientType.TO, new InternetAddress(email));
+            mimeMessage.setSubject(subject);
+            mimeMessage.setContent(message, "text/html");
+            javaMailSender.send(mimeMessage);
+        } catch (MessagingException e) {
+            String msg = "Sending message " + subject + " to email " + email + " was failed!";
+            LOGGER.error(msg, e);
+        }
+    }
```
Comment (MarrMika) [inline src/main/java/com/ita/if103java/ims/service/impl/MailServiceImpl.java]: This is unnecessary.  Use the method above.

File: src/main/java/com/ita/if103java/ims/dao/impl/AccountTypeDaoImpl.java [hunk 2]
```diff
@@ -170,5 +179,7 @@ private PreparedStatement createStatement(AccountType accountType, Connection co
         static final String SQL_SET_ACTIVE_STATUS_ACCOUNT_TYPE = "UPDATE account_types SET active = ? WHERE id = ?";
 
         static final String SQL_FIND_MIN_LVL_TYPE = "SELECT MIN(level) FROM account_types";
+
+        static final String SQL_FIND_ALL_POSSIBLE_TO_UPGRADE = "SELECT * FROM account_types WHERE level > ?";
```
Comment (ikrasWork) [inline src/main/java/com/ita/if103java/ims/dao/impl/AccountTypeDaoImpl.java]: I was thinking about this and decided that we need to select all. Please left this method for now, but most probably we will not use it and remove in future

**Comment by ikrasWork** [comment]
Please  resolve conflicts with and can merge

File: src/main/java/com/ita/if103java/ims/service/impl/InvitationServiceImpl.java [hunk 3]
```diff
@@ -3,40 +3,60 @@
...
-        if (allowToInvite(accountId))
-            sendInvitationMessage(email, message);
+    public void inviteUser(String email, Long accountId) {
+        if (allowToInvite(accountId)) {
+            User user = userDao.create(new User());
+            user.setEmail(email);
+            user.setRole(Role.WORKER);
+            user.setAccountId(accountId);
+            String emailUUID = UUID.randomUUID().toString();
+            user.setEmailUUID(emailUUID);
```
[the remark this answers is not in the document]
Comment (ikrasWork) [reply src/main/java/com/ita/if103java/ims/service/impl/InvitationServiceImpl.java]: I didn't see you changed this

**Pull request merged**